hack/local-up-cluster.sh

```diff
@@ -16,7 +16,9 @@
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 
-# This script builds and runs a local kubernetes cluster. You may need to run
+#get the default gateway
+DEFAULT_GATEWAY=$(ip route get 1.1.1.1 | grep -Po '(?<=(src )).*(?= uid)')
```

Comment:
what if we also default to localhost in case that the command fails, so we are confident we don't break possible scenarios without a default route
```
DEFAULT_HOSTNAME=${DEFAULT_GATEWAY:-"127.0.0.1"}
```

and `HOSTNAME_OVERRIDE=${HOSTNAME_OVERRIDE:-$DEFAULT_HOSTNAME}`